### PR TITLE
doc: implement semaphoreKeyPts

### PR DESCRIPTION
**Pull request by martinvibes**
## 📝 Summary
implement SemaphoreVerifierKeyPts contract to Soroban #1

## ✅ Self-Check for Contribution Requirements
Mark with an `x` all the checkboxes that apply (like `[x]`).
- [x] Forked the repository and created a new branch with a meaningful name.
- [x] Ensured that the content is clear and free of grammatical errors.
- [x] Maintained consistency in tone and formatting throughout the document.
- [x] Verified that all links work correctly and checked for typos before submitting the changes.
- [x] Used the conventional commit format for commit messages.
- [x] Thoroughly tested changes, including unit tests if applicable.

## 💬 Comments
Any additional context, questions, or considerations for the reviewers.

### CODE REVIEW

**Comment by martinvibes** [comment]
hello @evgongora kindly review :)

**Comment by martinvibes** [comment]
hello @evgongora kindly review :)

File: research/SemaphoreKeyPts-implementation.md [hunk 1]
```diff
@@ -0,0 +1,453 @@
...
+### Key Features
+
+- Verification key point storage and management
+- Merkle tree depth-based point retrieval
+- Data structure validation
+- Robust error handling
+
+## Installation Guide & Setup
+
+### Project Setup
```
Comment (evgongora) [inline research/SemaphoreKeyPts-implementation.md]: This is not necessary please remove Project setup section.

File: research/SemaphoreKeyPts-implementation.md [hunk 2]
```diff
@@ -0,0 +1,451 @@
...
+```bash
+# Create project directory
+mkdir semaphore-project
+cd semaphore-project
+
+# Initialize new Rust project
+cargo new semaphore-contract --lib
+cd semaphore-contract
+```
+
```
Comment (evgongora) [inline research/SemaphoreKeyPts-implementation.md]: This is not necessary.

**Comment by martinvibes** [comment]
@evgongora removed requested changes

**Comment by evgongora** [comment]
@martinvibes Could you please remove excessive comments for basic Rust or Soroban concepts? (e.g., what #![no_std] means, how #[contract] works)
Let's focus on the key concepts for the migration of the contract.

**Comment by martinvibes** [comment]
@evgongora kindly review :)

**Comment by evgongora** [comment]
LGTM, great job!

**Pull request merged**